File: app/src/main/java/com/adisa/diningplus/adapters/NutritionAdapter.java
package com.adisa.diningplus.adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.adisa.diningplus.R;
import com.adisa.diningplus.activities.ItemActivity;

import java.util.ArrayList;

public class NutritionAdapter extends BaseAdapter {
    private Context context;
    private ArrayList<ItemActivity.Nutrient> nutrients;

    class ViewHolder {
        TextView name;
        TextView amount;
        TextView pdv;
    }

    public NutritionAdapter(Context context, ArrayList<ItemActivity.Nutrient> nutrients) {
        this.context = context;
        this.nutrients = nutrients;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        ViewHolder viewHolder;
        ItemActivity.Nutrient allergen = nutrients.get(position);
        if (convertView == null) {
            viewHolder = new ViewHolder();
            convertView = inflater.inflate(R.layout.nutrient, null);

            viewHolder.name = (TextView) convertView.findViewById(R.id.nutrient_name);
            viewHolder.amount = (TextView) convertView.findViewById(R.id.nutrient_amount);
            viewHolder.pdv = (TextView) convertView.findViewById(R.id.nutrient_pdv);

            convertView.setEnabled(false);
            convertView.setOnClickListener(null);
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
        }
        viewHolder.name.setText(allergen.name);
        viewHolder.amount.setText(allergen.amount);
        if (allergen.pdv != null) {
            viewHolder.pdv.setText(allergen.pdv + "%");
        } else {
            viewHolder.pdv.setText("");
        }

        return convertView;
    }

    @Override
    public int getCount() {
        return nutrients.size();
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }
}

File: app/src/main/java/com/adisa/diningplus/adapters/NutritionAdapter.kt
package com.adisa.diningplus.adapters

import android.content.Context
import android.graphics.Typeface
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView
import com.adisa.diningplus.R
import com.adisa.diningplus.activities.ItemActivity
import com.adisa.diningplus.activities.ItemActivity.Nutrient
import java.util.*

class NutritionAdapter(private val context: Context, private val nutrients: ArrayList<Nutrient>) : BaseAdapter() {

    internal inner class ViewHolder {
        var indent: View? = null
        var name: TextView? = null
        var amount: TextView? = null
        var pdv: TextView? = null
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var convertView = convertView
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val viewHolder: ViewHolder
        val nutrient = nutrients[position]
        if (convertView == null) {
            viewHolder = ViewHolder()
            convertView = inflater.inflate(R.layout.nutrient, null)
            viewHolder.indent = convertView.findViewById(R.id.nutrient_indent)
            viewHolder.name = convertView.findViewById(R.id.nutrient_name)
            viewHolder.amount = convertView.findViewById(R.id.nutrient_amount)
            viewHolder.pdv = convertView.findViewById(R.id.nutrient_pdv)
            convertView.isEnabled = false
            convertView.setOnClickListener(null)
            convertView.tag = viewHolder
        } else {
            viewHolder = convertView.tag as ViewHolder
        }
        viewHolder.indent!!.visibility =  if (nutrient.type == ItemActivity.NutrientType.SUB) View.VISIBLE else View.GONE
        viewHolder.name!!.text = nutrient.name
        if (nutrient.type == ItemActivity.NutrientType.SUB || nutrient.type == ItemActivity.NutrientType.PLAIN) {
            viewHolder.name!!.typeface = Typeface.DEFAULT
        }
        viewHolder.amount!!.text = nutrient.amount
        if (nutrient.pdv != null) {
            viewHolder.pdv!!.text = nutrient.pdv.toString() + "%"
        } else {
            viewHolder.pdv!!.text = ""
        }
        return convertView!!
    }

    override fun getCount(): Int {
        return nutrients.size
    }

    override fun getItem(position: Int): Any? {
        return null
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

}